normandy/recipes/migrations/0014_auto_20190228_1128.py

```diff
@@ -0,0 +1,85 @@
+# Generated by Django 2.0.13 on 2019-02-28 11:28
+
+import json
+
+from urllib.parse import unquote_plus
+
+from django.db import migrations
+
+
+def remove_signatures(apps, schema_editor):
+    Recipe = apps.get_model("recipes", "Recipe")
+    Action = apps.get_model("recipes", "Action")
+    Signature = apps.get_model("recipes", "Signature")
+
+    for recipe in Recipe.objects.exclude(signature=None):
+        sig = recipe.signature
+        recipe.signature = None
+        recipe.save()
+        sig.delete()
+
+    for action in Action.objects.exclude(signature=None):
+        sig = action.signature
+        action.signature = None
+        action.save()
+        sig.delete()
+
+    for sig in Signature.objects.all():
+        sig.delete()
+
+
+def add_extension_id(apps, schema_editor):
+    Action = apps.get_model("recipes", "Action")
+    RecipeRevision = apps.get_model("recipes", "RecipeRevision")
+    Extension = apps.get_model("studies", "Extension")
+
+    try:
+        action = Action.objects.get(name="opt-out-study")
+    except Action.DoesNotExist:
+        return  # Do nothing since there cannot be any recipes using the opt-out-study action
+
+    revisions = RecipeRevision.objects.filter(action_id=action.id)
+
+    for revision in revisions:
+        arguments = json.loads(revision.arguments_json)
+        url = arguments.get("addonUrl")
+        filename = unquote_plus(url.split("/extensions/").pop())
+        extension = Extension.objects.get(xpi=f"extensions/{filename}")
+        arguments["extensionApiId"] = extension.id
```

Comment:
@peterbe pointed out in another PR when I was doing something similar that this method of choosing add-ons is a bit weak. I wanted to share some code I wrote that did the same thing slightly differently. Specifically it parsed the URL and grabbed the path, and then split by slashes. That helps with potential query strings, even though that's unlikely.

https://github.com/mozilla/normandy/blob/5718076dcf8dd79caf0a4d8a2ab0f618276e906c/normandy/recipes/management/commands/update_addon_urls.py